src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -87,7 +87,14 @@ func (client *Client) targetQOS(req *http.Request) int {
 // responsible for responding to the initial CONNECT request with a 200 OK.
 func pipeData(clientConn net.Conn, connOut net.Conn, req *http.Request) {
 	// Start piping from client to proxy
-	go io.Copy(connOut, clientConn)
+	go func() {
+		io.Copy(connOut, clientConn)
+		// Force closing if EOF at the request half or error encountered.
+		// A bit arbitrary, but it's rather rare now to use half closing
+		// as a way to notify server. Most application closes both connections
+		// after completed send / receive so that won't cause problem.
+		clientConn.Close()
```

Comment:
This seems like it only makes sense if the ```io.Copy(connOut, clientConn)``` call returns an error no? If it's just returning because of EOF on clientConn, that means clientConn is already closed, so no need to close it again. 

It seems like we might as well also call connOut.Close here as well, as there's no clientConn for connOut to write to at that point and it seems like there are cases where we'd otherwise rely on the server connection timing out (i.e. if the clientConn hasn't written any data and has just closed, the server connection could be waiting around for data up until some server-side timeout).

Does that make sense?